discord/http.py

```diff
@@ -1058,6 +1070,63 @@ def unpin_message(self, channel_id: Snowflake, message_id: Snowflake, reason: Op
     def pins_from(self, channel_id: Snowflake) -> Response[List[message.Message]]:
         return self.request(Route('GET', '/channels/{channel_id}/pins', channel_id=channel_id))
 
+    async def send_voice_message(self, channel_id: Snowflake, VoiceMessage: VoiceMessageFile):
+        """|coro|
+
+        Sends a voice message to the specified channel.
+
+        Parameters
+        -----------
+        channel_id: :class:`~discord.abc.Snowflake`
+            The ID of the channel to send the voice message to.
+        VoiceMessage: :class:`~discord.VoiceMessageFile`
+            The voice message file to send. This should be an instance of :class:`~discord.VoiceMessageFile`.
+        """
+        from .message import MessageFlags
+        uploadRoute = Route('POST', '/channels/{channel_id}/attachments', channel_id=channel_id)
+        payload = {
+            "files": [{
+                "filename": "voice-message.ogg",
+                "file_size": VoiceMessage.size(),
+                "id": 0,
+            }]
+        }
+        response = await self.request(uploadRoute, json=payload)
+
+        upload_data = response['attachments'][0]
+        upload_url = upload_data["upload_url"]
+        uploaded_filename = upload_data["upload_filename"]
+
+        import requests
+
+        t = requests.put(upload_url, headers={"Content-Type": "audio/ogg"}, data=VoiceMessage.fp)
+        print(f"Status code: {t.status_code}")
+
+        # x = await self.__session.request("PUT", upload_url, headers={"Content-Type": "audio/ogg"}, data=VoiceMessage.fp)
+        # print("*********")
+        # print(upload_url)
+        # print(x.read())
+        # print("*********")
+
+        VoiceMessage.uploaded_filename = uploaded_filename
+
+        r = Route('POST', '/channels/{channel_id}/messages', channel_id=channel_id)
+
+        message_payload = {
+            "flags": 8192,  # IS_VOICE_MESSAGE
+            "attachments": [VoiceMessage.to_dict(0)],
+        }
+
+        headers = {"Authorization": f"Bot {self.token}",
+                   "Content-Type": "application/json"}
+
+        res = requests.post("" + r.url, headers=headers, json=message_payload)
+        return res.json()
```

Comment:
Why are we going from using the library provided (and correct) http request format, into import requests (a non-required discord.py dependency, and sync/blocking library) for an entirely different route?
Not to mention this code is a mish-mash of differing quality.